File: pytools/examples/golang/src/server/proto/msg_scene_player.go
package proto

import (
	"packet"
)

type MsgScenePlayer struct {
	uid                      uint32
	sceneRotPos              *MsgSceneRotPos
}

func (this *MsgScenePlayer) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint32(this.uid)
	pack.WriteBytes(this.sceneRotPos.Encode())

	return pack.ReadBytes()
}

func MsgScenePlayerDecode(pack *packet.Packet) *MsgScenePlayer {
	msgScenePlayer := &MsgScenePlayer{}

	msgScenePlayer.uid = pack.ReadUint32()
	msgScenePlayer.sceneRotPos = MsgSceneRotPosDecode(pack)
	return msgScenePlayer
}

func (this *MsgScenePlayer) SetUid(uid uint32) {
	this.uid = uid
}

func (this *MsgScenePlayer) GetUid() uint32 {
	return this.uid
}

func (this *MsgScenePlayer) SetSceneRotPos(sceneRotPos *MsgSceneRotPos) {
	this.sceneRotPos = sceneRotPos
}

func (this *MsgScenePlayer) GetSceneRotPos() *MsgSceneRotPos {
	return this.sceneRotPos
}

File: pytools/data/data.code.csharp.client/MsgScenePlayer.cs
using System.Collections;
using System.Collections.Generic;


public class MsgScenePlayer
{
	private uint _uid;
	private MsgSceneRotPos _scene_rot_pos;


	public Packet Encode()
	{
		Packet packet = new Packet();
		packet.WriteUint(this._uid);
		packet.WriteBuffer(this._scene_rot_pos.GetBuffer());
		return packet;
	}

	public MsgScenePlayer()
	{
	}

	public MsgScenePlayer(Packet packet)
	{
		this._uid = packet.ReadUint();
		this._scene_rot_pos = new MsgSceneRotPos(packet);
	}

	public List<byte> GetBuffer()
	{
		return this.Encode().GetBuffer();
	}


	public uint uid
	{
		get { return this._uid; }
		set { this._uid = value; }
	}

	public MsgSceneRotPos scene_rot_pos
	{
		get { return this._scene_rot_pos; }
		set { this._scene_rot_pos = value; }
	}

}
